cfme/fixtures/vm.py

```diff
@@ -68,3 +70,33 @@ def full_template_vm_modscope(setup_provider_modscope, request, full_template_mo
 def full_template_vm(setup_provider, request, full_template, provider):
     vm_name = _get_vm_name(request)
     return _create_vm(request, full_template, provider, vm_name)
+
+
+def _create_instance(appliance, provider, template_name):
+    instance = appliance.collections.cloud_instances.instantiate(random_vm_name('pwr-c'),
+                                                                 provider,
+                                                                 template_name)
+    if not instance.exists_on_provider:
+        instance.create_on_provider(allow_skip="default", find_in_cfme=True)
+    elif instance.provider.one_of(EC2Provider) and instance.mgmt.state == VmState.DELETED:
+        instance.mgmt.rename('test_terminated_{}'.format(fauxfactory.gen_alphanumeric(8)))
+        instance.create_on_provider(allow_skip="default", find_in_cfme=True)
+    return instance
+
+
+@pytest.fixture(scope="function")
+def testing_instance(appliance, provider, small_template, setup_provider):
```

Comment:
I think we do not need 2 fixtures, we can create and yield two instances from single fixture. Please check if that works for your tests.